bdsx/bds/blockpos.ts

```diff
@@ -371,8 +371,19 @@ export class Vec3 extends NativeStruct {
     toJSON():VectorXYZ {
         return {x:this.x, y:this.y, z:this.z};
     }
+
+    directionFromRotation(rotation:Vec2):Vec3 {
+        abstract();
+    }
+
+    rotationFromDirection(direction:Vec3):Vec2 {
+        abstract();
+    }
 }
 
+Vec3.prototype.directionFromRotation = procHacker.js('?directionFromRotation@Vec3@@SA?AV1@AEBVVec2@@@Z', Vec3, {this: Vec3}, Vec2);
```

Comment:
it doesn't need to set `this` option because they are static methods.
but they return structures so you need to set `structureReturn: true`.

it seems that your test didn't fail because `this` and `structureReturn` both allocate new spaces.